addons/hide-flyout/style.css

```diff
@@ -1,80 +1,80 @@
 }
 
-.injectionDiv {
-  border-left-color: transparent;
-}
-
-.gui[dir="rtl"] .injectionDiv {
-  border-left-color: var(--editorDarkMode-border, rgba(0, 0, 0, 0.15));
-  border-right-color: transparent;
-}
-
-.injectionDiv::after {
-  /* left border of the code area:
-     must be above the flyout and below the extension button */
-  content: "";
-  display: block;
+/* The default left-side border does not work properly when a block is being dragged, */
+/* which lets a 1 pixel column of the flyout be visible. */
+/* To fix this we use two elements that are only visible when a block is being dragged. */
+/* The first element is white/black to cover up the contents, the other has the correct border color. */
+/* Two elements are necessary because the border color has transparency. */
+.sa-flyout-border-1,
+.sa-flyout-border-2 {
   position: absolute;
+  /* above flyout but below add extension button */
   z-index: 40;
   top: 0;
   bottom: 8px;
   left: -1px;
-  border-left: 1px solid var(--editorDarkMode-border, rgba(0, 0, 0, 0.15));
 }
-
-.gui[dir="rtl"] .injectionDiv::after {
+[dir="rtl"] .sa-flyout-border-1,
+[dir="rtl"] .sa-flyout-border-2 {
   left: auto;
   right: -1px;
 }
+.sa-flyout-border-1 {
+  border-left: 1px solid var(--editorDarkMode-page, hsla(215, 100%, 95%, 1));
+}
+.sa-flyout-border-2 {
+  border-left: 1px solid var(--editorDarkMode-border, rgba(0, 0, 0, 0.15));
+}
 
 .blocklyFlyout,
-.blocklyFlyoutScrollbar,
-.sa-lock-image {
+.blocklyFlyoutScrollbar {
   transition-property: margin;
 }
 
-.gui[dir="ltr"] .blocklyFlyout.sa-flyoutClose,
-.gui[dir="ltr"] .blocklyFlyoutScrollbar.sa-flyoutClose,
-.gui[dir="ltr"] .sa-lock-image.sa-flyoutClose {
+[dir="ltr"] .blocklyFlyout.sa-flyoutClose,
+[dir="ltr"] .blocklyFlyoutScrollbar.sa-flyoutClose {
   margin-left: calc(0px - var(--sa-flyout-width) - 10px);
 }
 
-.gui[dir="rtl"] .blocklyFlyout.sa-flyoutClose,
-.gui[dir="rtl"] .blocklyFlyoutScrollbar.sa-flyoutClose {
+[dir="rtl"] .blocklyFlyout.sa-flyoutClose,
+[dir="rtl"] .blocklyFlyoutScrollbar.sa-flyoutClose {
   margin-left: calc(var(--sa-flyout-width) + 10px);
 }
 
-.gui[dir="rtl"] .sa-lock-image.sa-flyoutClose {
-  margin-right: calc(0px - var(--sa-flyout-width) - 10px);
-}
-
 .sa-flyout-placeHolder {
   display: var(--hideFlyout-placeholderDisplay) !important;
   position: absolute;
-  top: 0;
-  left: calc(var(--sa-category-width) + 1px);
   height: 100%;
   width: calc(var(--sa-flyout-width) + 1px);
+  top: 0;
+}
+
+[dir="ltr"] .sa-flyout-placeHolder {
+  left: calc(var(--sa-category-width) + 1px);
 }
 
 [dir="rtl"] .sa-flyout-placeHolder {
-  left: auto;
   right: calc(var(--sa-category-width) + 1px);
 }
 
-.sa-lock-image {
+.sa-lock-object {
   display: var(--hideFlyout-lockDisplay) !important;
-  position: absolute;
-  z-index: 20;
-  cursor: pointer;
-  top: 4px;
-  left: calc(var(--sa-toolbox-width) - 15px - 32px);
+  transform: translate(calc(var(--sa-flyout-width) - 15px - 32px), 4px);
```

Comment:
The previous 4px vertical offset included the top border of the `.injectionDiv`, so this should now be 3px.